IndexOutOfBoundsException whilst GUI with no slots is open

MinecraftForge: 1.12.2-14.23.5.2770
AutoRegLib: 1.3-20
Psi: r1.1-59

When a CAD is in a players inventory (*not* in the hotbar) and a container GUI is opened that has no slots (tested with the GUIs of Super Sound Muffler and Industrial Foregoing whilst configuring machine sides), the log is spammed with following stacktrace (and if you have VanillaFix installed, the game lags out and a seemingly endless amount of error message toasts from that mod pop up):
```
[14:24:41] [main/FATAL] [net.minecraft.client.Minecraft]: Error executing task
java.util.concurrent.ExecutionException: java.lang.IndexOutOfBoundsException: Index: 9, Size: 0
	at java.util.concurrent.FutureTask.report(Unknown Source) ~[?:1.8.0_181]
	at java.util.concurrent.FutureTask.get(Unknown Source) ~[?:1.8.0_181]
	at net.minecraft.util.Util.func_181617_a(SourceFile:47) [h.class:?]
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1087) [bib.class:?]
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:397) [bib.class:?]
	at net.minecraft.client.main.Main.main(SourceFile:123) [Main.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_181]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_181]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_181]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_181]
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.12.jar:?]
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_181]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_181]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_181]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_181]
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:196) [NewLaunch.jar:?]
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:231) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.listen(EntryPoint.java:143) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.main(EntryPoint.java:34) [NewLaunch.jar:?]
Caused by: java.lang.IndexOutOfBoundsException: Index: 9, Size: 0
	at java.util.ArrayList.rangeCheck(Unknown Source) ~[?:1.8.0_181]
	at java.util.ArrayList.get(Unknown Source) ~[?:1.8.0_181]
	at net.minecraft.inventory.Container.func_75139_a(Container.java:120) ~[afr.class:?]
	at net.minecraft.inventory.Container.func_75141_a(Container.java:519) ~[afr.class:?]
	at net.minecraft.client.network.NetHandlerPlayClient.func_147266_a(NetHandlerPlayClient.java:1154) ~[brz.class:?]
	at net.minecraft.network.play.server.SPacketSetSlot.func_148833_a(SourceFile:30) ~[iu.class:?]
	at net.minecraft.network.play.server.SPacketSetSlot.func_148833_a(SourceFile:10) ~[iu.class:?]
	at net.minecraft.network.PacketThreadUtil$1.run(SourceFile:13) ~[hv$1.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[?:1.8.0_181]
	at java.util.concurrent.FutureTask.run(Unknown Source) ~[?:1.8.0_181]
	at net.minecraft.util.Util.func_181617_a(SourceFile:46) ~[h.class:?]
	... 17 more
```

Here's the full latest.log: https://gist.githubusercontent.com/SanAndreasP/3ee93b2294746d1e5611c21294eec98a/raw/be686b7cc38a9ba0371dfa42ad1030acfaa56139/latest.log

--- a/src/main/java/vazkii/psi/api/PsiAPI.java
+++ b/src/main/java/vazkii/psi/api/PsiAPI.java
@@ -151,4 +151,15 @@ public final class PsiAPI {
 		return slot;
 	}
 
+	public static boolean canCADBeUpdated(EntityPlayer player) {
+		if(player == null)
+			return false;
+
+		if(player.openContainer == null)
+			return true;
+
+		int cadSlot = getPlayerCADSlot(player);
+		return cadSlot < 9 || cadSlot == 40;
+	}
+
 }

--- a/src/main/java/vazkii/psi/common/core/handler/PlayerDataHandler.java
+++ b/src/main/java/vazkii/psi/common/core/handler/PlayerDataHandler.java
@@ -169,7 +169,7 @@ public class PlayerDataHandler {
 				EntityPlayer player = (EntityPlayer) event.getEntityLiving();
 				
 				ItemStack cadStack = PsiAPI.getPlayerCAD(player);
-				if(!cadStack.isEmpty() && cadStack.getItem() instanceof ICAD)
+				if(!cadStack.isEmpty() && cadStack.getItem() instanceof ICAD && PsiAPI.canCADBeUpdated(player))
 					((ICAD) cadStack.getItem()).incrementTime(cadStack);
 				
 				PsiArmorEvent.post(new PsiArmorEvent(player, PsiArmorEvent.TICK));

